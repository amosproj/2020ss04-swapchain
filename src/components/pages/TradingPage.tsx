import React, { useState } from "react"
import { Link } from "react-router-dom"
import { ReactComponent as Key } from "../../icons/key.svg"
import { ReactComponent as LockClosed } from "../../icons/lock-closed.svg"
import { ReactComponent as Collection } from "../../icons/collection.svg"
import { RadioButton } from "../forms/RadioButton"
import { Form } from "../forms/Form"

export const TradingPage = () => {
  enum Operation {
    ACCEPT,
    PROPOSE,
  }

  const [operation, setOperation] = useState(Operation.PROPOSE)

  return (
    <div className="flex flex-col items-center justify-between mt-12 space-y-8 rounded md:space-x-8 md:flex-row md:space-y-0">
      <div className="md:w-8/12">
        <h2 className="mt-2 text-3xl leading-8 tracking-tight text-gray-900 sm:text-4xl sm:leading-10">
          Our mission
        </h2>
        <p className="mt-4 leading-7 text-gray-700">
    Swapchainâ€™s mission is to create an environment to carry out Atomic Cross-Chain Swaps (ACCS).<br></br>
    The core functionality is the execution of over-the-counter transactions between the Bitcoin and the Bitshares blockchain while making use of Hash Time Locked Contracts (HTLCs).
        </p>
        <ul className="flex flex-col mt-8 space-y-4">
          <li className="flex-grow">
            <div className="flex">
              <div className="flex-shrink-0">
                <Key className="flex items-center justify-center w-8 text-teal-500"></Key>
              </div>
              <div className="ml-4">
                <h4 className="text-lg leading-6 text-gray-600">You own your private keys</h4>
              </div>
            </div>
          </li>
          <li>
            <div className="flex">
              <div className="flex-shrink-0">
                <LockClosed className="flex items-center justify-center w-8 text-teal-500"></LockClosed>
              </div>
              <div className="ml-4">
                <h4 className="text-lg leading-6 text-gray-600">Zero trust required</h4>
              </div>
            </div>
          </li>
          <li>
            <div className="flex">
              <div className="flex-shrink-0">
                <Collection className="flex items-center justify-center w-8 text-teal-500"></Collection>
              </div>
              <div className="ml-4">
                <h4 className="text-lg leading-6 text-gray-600">Trade crypto assets the way you like</h4>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div className="md:w-4/12">
        <Form
          main={
            <div className="flex flex-col justify-center w-full p-4 space-y-4">
              <div>
                <h2 className="text-xl leading-tight text-gray-900">Start trading</h2>
                <p className="mt-1 text-sm leading-4 text-gray-600">
                  Create a new ACCS or accept an existing proposal from your trading partner.
                </p>
              </div>
              <div className="flex flex-col space-y-4 xl:flex-row xl:space-y-0 xl:space-x-4">
                <RadioButton
                  description="a new Atomic Cross Chain Swap"
                  name="Propose"
                  onClick={() => setOperation(Operation.PROPOSE)}
                  selected={operation === Operation.PROPOSE}
                ></RadioButton>
                <RadioButton
                  description="an Atomic Cross Chain Swap proposal"
                  name="Accept"
                  onClick={() => setOperation(Operation.ACCEPT)}
                  selected={operation === Operation.ACCEPT}
                ></RadioButton>
              </div>
            </div>
          }
          footer={
            <Link
              to={operation === Operation.ACCEPT ? "/accept" : "/propose"}
              className="px-8 py-3 font-semibold text-gray-900 uppercase transition duration-200 ease-in-out bg-white border border-teal-400 rounded-sm shadow-sm focus:outline-none hover:border-teal-500 hover:bg-white"
            >
              Start
            </Link>
          }
        ></Form>
      </div>
    </div>
  )
}
