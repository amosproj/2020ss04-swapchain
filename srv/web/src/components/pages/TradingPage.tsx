import React, { useState } from "react"
import { Link } from "react-router-dom"
import { ReactComponent as EyeOff } from "../../icons/eye-off.svg"
import { ReactComponent as Annotation } from "../../icons/annotation.svg"
import { ReactComponent as QRCode } from "../../icons/qrcode.svg"
import { RadioButton } from "../forms/RadioButton"
import { Form } from "../forms/Form"

export const TradingPage = () => {
    
    enum Operation {
        ACCEPT,
        PROPOSE,
    }



    const [operation, setOperation] = useState(Operation.PROPOSE)






    return (
        <div className="flex flex-col items-center justify-between mt-12 space-y-8 rounded md:space-x-8 md:flex-row md:space-y-0">
            <div className="flex-grow">
                <h2 className="mt-2 text-3xl leading-8 tracking-tight text-gray-900 sm:text-4xl sm:leading-10">
                    Atomic Cross Chain Swaps
                </h2>
                <p className="mt-4 leading-7 text-gray-500">
                    No more enterprise wallets saving your confidential data and contradicting the idea behind distributed ledger. 
                    No more transactions with mediocre safety measures leaving you worried about fraudulent exchange partners.
                    Swapchain is here to combine trustless safety measures with user data privacy. Because cryptocurrency trading deserves better.

                </p>
                <ul className="flex flex-col mt-8 space-y-4">
                    <li className="flex-grow">
                        <div className="flex">
                            <div className="flex-shrink-0">
                                <div className="flex items-center justify-center w-8 text-teal-500">
                                    <EyeOff></EyeOff>
                                </div>
                            </div>
                            <div className="ml-4">
                                <h4 className="text-lg leading-6 text-gray-900">The important stuff stays with you. </h4>
                                <p className="mt-2 text-base leading-6 text-gray-500">
                                    Swapchain lets you keep your public keys, private keys and addresses in your browser exclusively. Nothing is stored in an external database, giving you full control over your confidential data.
                                    </p>
                            </div></div>
                    </li>
                    <li>
                        <div className="flex">
                            <div className="flex-shrink-0">
                                <div className="flex items-center justify-center w-8 text-teal-500">
                                    <Annotation></Annotation>
                                </div>
                            </div>
                            <div className="ml-4">
                                <h4 className="text-lg leading-6 text-gray-900">No need to trust your exchange partner. </h4>
                                <p className="mt-2 text-base leading-6 text-gray-500">
                                    As Swapchain uses hash-time-lock-contracts (HTLCs), counterparty risks are minimized. This means you don't have to trust your exchange partner and still can trade safely.
                                    </p>
                            </div></div>
                    </li>
                    <li>
                        <div className="flex">
                            <div className="flex-shrink-0">
                                <div className="flex items-center justify-center w-8 text-teal-500">
                                    <QRCode></QRCode>
                                </div>
                            </div>
                            <div className="ml-4">
                                <h4 className="text-lg leading-6 text-gray-900">Trade crypto assets the way you like it. </h4>
                                <p className="mt-2 text-base leading-6 text-gray-500">
                                    With Swapchain, you can choose your preferred transaction duration and time-lock length, making your cryptocurrency transactions both more flexible to your needs and safer.
                                </p>
                            </div></div>
                    </li>
                </ul>

            </div>
            <Form main={
                <div className="flex flex-col justify-center w-full p-8 space-y-4">
                    <div>
                        <h2 className="text-xl leading-tight text-gray-900">Start trading</h2>
                        <p className="mt-1 text-sm leading-4 text-gray-600">Create a new ACCS or accept an existing proposal from your trading partner.</p>
                    </div>
                    <div className="flex flex-col space-y-4 lg:flex-row lg:space-y-0 lg:space-x-4">

                        <RadioButton
                            description="a new Atomic Cross Chain Swap"
                            name="Propose"
                            onClick={() => setOperation(Operation.PROPOSE)}
                            selected={operation === Operation.PROPOSE}
                        ></RadioButton>
                        <RadioButton
                            description="an Atomic Cross Chain Swap proposal"
                            name="Accept"
                            onClick={() => setOperation(Operation.ACCEPT)}
                            selected={operation === Operation.ACCEPT}
                        ></RadioButton>
                    </div>
                </div>
            }
                footer={
                    <Link to={operation === Operation.ACCEPT ? "/accept" : "/propose"} className="px-8 py-3 text-gray-900 uppercase transition duration-200 ease-in-out bg-white border rounded-sm shadow-sm focus:outline-none hover:border-teal-500 hover:bg-white">
                        Start
</Link>
                }
            ></Form>
        </div>
    )
}
