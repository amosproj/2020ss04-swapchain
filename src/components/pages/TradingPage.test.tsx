import React from "react"
import { TradingPage } from "./TradingPage"
import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom"

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <TradingPage></TradingPage>
      </BrowserRouter>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
