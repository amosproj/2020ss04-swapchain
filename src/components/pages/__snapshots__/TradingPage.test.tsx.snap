// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col items-center justify-between mt-12 space-y-8 rounded md:space-x-8 md:flex-row md:space-y-0"
>
  <div
    className="flex-grow"
  >
    <h2
      className="mt-2 text-3xl leading-8 tracking-tight text-gray-900 sm:text-4xl sm:leading-10"
    >
      Atomic Cross Chain Swaps
    </h2>
    <p
      className="mt-4 leading-7 text-gray-500"
    >
      Even more Explanation Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
    </p>
    <ul
      className="flex flex-col mt-8 space-y-4"
    >
      <li
        className="flex-grow"
      >
        <div
          className="flex"
        >
          <div
            className="flex-shrink-0"
          >
            <div
              className="flex items-center justify-center w-8 text-teal-500"
            >
              <svg>
                eye-off.svg
              </svg>
            </div>
          </div>
          <div
            className="ml-4"
          >
            <h4
              className="text-lg leading-6 text-gray-900"
            >
              It's super cool
            </h4>
            <p
              className="mt-2 text-base leading-6 text-gray-500"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
            </p>
          </div>
        </div>
      </li>
      <li>
        <div
          className="flex"
        >
          <div
            className="flex-shrink-0"
          >
            <div
              className="flex items-center justify-center w-8 text-teal-500"
            >
              <svg>
                annotation.svg
              </svg>
            </div>
          </div>
          <div
            className="ml-4"
          >
            <h4
              className="text-lg leading-6 text-gray-900"
            >
              Zero trust required
            </h4>
            <p
              className="mt-2 text-base leading-6 text-gray-500"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
            </p>
          </div>
        </div>
      </li>
      <li>
        <div
          className="flex"
        >
          <div
            className="flex-shrink-0"
          >
            <div
              className="flex items-center justify-center w-8 text-teal-500"
            >
              <svg>
                qrcode.svg
              </svg>
            </div>
          </div>
          <div
            className="ml-4"
          >
            <h4
              className="text-lg leading-6 text-gray-900"
            >
              Because..
            </h4>
            <p
              className="mt-2 text-base leading-6 text-gray-500"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="flex flex-col items-center w-full bg-white border rounded shadow-lg"
  >
    <div
      className="flex flex-col justify-center w-full p-4 space-y-4"
    >
      <div
        className="flex flex-col justify-center w-full p-8 space-y-4"
      >
        <div>
          <h2
            className="text-xl leading-tight text-gray-900"
          >
            Start trading
          </h2>
          <p
            className="mt-1 text-sm leading-4 text-gray-600"
          >
            Create a new ACCS or accept an existing proposal from your trading partner.
          </p>
        </div>
        <div
          className="flex flex-col space-y-4 lg:flex-row lg:space-y-0 lg:space-x-4"
        >
          <button
            className="focus:outline-none flex w-full items-center justify-between border rounded border-teal-400 shadow hover:bg-white hover:shadow-sm"
            onClick={[Function]}
          >
            <div
              className="flex flex-col items-start justify-between p-4"
            >
              <div
                className="flex items-center justify-between w-full"
              >
                <span
                  className="text-left font-semibold text-gray-900"
                >
                  Propose
                </span>
                <span
                  className="text-xs font-thin text-gray-600"
                />
              </div>
              <p
                className="mt-2 text-sm text-left text-gray-600"
              >
                a new Atomic Cross Chain Swap
              </p>
            </div>
            <div
              className="flex items-center justify-center"
            >
              <span
                className="mr-4 text-3xl font-semibold hidden md:block text-gray-900 "
              />
            </div>
          </button>
          <button
            className="focus:outline-none flex w-full items-center justify-between border rounded bg-gray-100 hover:bg-white hover:shadow-sm"
            onClick={[Function]}
          >
            <div
              className="flex flex-col items-start justify-between p-4"
            >
              <div
                className="flex items-center justify-between w-full"
              >
                <span
                  className="text-left font-semibold text-gray-600"
                >
                  Accept
                </span>
                <span
                  className="text-xs font-thin text-gray-400"
                />
              </div>
              <p
                className="mt-2 text-sm text-left text-gray-400"
              >
                an Atomic Cross Chain Swap proposal
              </p>
            </div>
            <div
              className="flex items-center justify-center"
            >
              <span
                className="mr-4 text-3xl font-semibold hidden md:block text-gray-300"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
    <footer
      className="flex justify-center w-full p-4 bg-gray-100 border-t border-gray-200"
    >
      <a
        className="px-8 py-3 text-gray-900 uppercase transition duration-200 ease-in-out bg-white border rounded-sm shadow-sm focus:outline-none hover:border-teal-500 hover:bg-white"
        href="/propose"
        onClick={[Function]}
      >
        Start
      </a>
    </footer>
  </div>
</div>
`;
